### PR TITLE
SDIT-1096 non-association reconciliation process

### PR DESCRIPTION
**Pull request by steverendell**
Relax comparison to ignore dps reason,
decrease the page size to 100,
avoid swapped NA details with same date

### CODE REVIEW

File: helm_deploy/hmpps-prisoner-to-nomis-update/templates/report-non-associations-cronjob.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   labels:
     {{- include "app.labels" . | nindent 4 }}
 spec:
-  schedule: "0 5 * * 1"
+  schedule: "10 4 * * 2"
```
Comment (steverendell) [inline helm_deploy/hmpps-prisoner-to-nomis-update/templates/report-non-associations-cronjob.yaml:8]: run on a tuesday instead to give it enough time

File: helm_deploy/hmpps-prisoner-to-nomis-update/templates/report-incentives-cronjob.yaml [hunk 2]
```diff
@@ -5,7 +5,7 @@ metadata:
   labels:
     {{- include "app.labels" . | nindent 4 }}
 spec:
-  schedule: "0 4 * * 1"
+  schedule: "10 4 * * 1"
```
Comment (steverendell) [inline helm_deploy/hmpps-prisoner-to-nomis-update/templates/report-incentives-cronjob.yaml:8]: avoid clash with the activities batch at 5:00 BST

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt [hunk 3]
```diff
@@ -271,7 +283,7 @@ class NonAssociationsReconciliationService(
     dpsReasonEnum: NonAssociation.Reason,
   ): Boolean {
     val t = translateToRolesAndReason(reason, recipReason)
-    return t.first != role1 || t.second != role2 || ((reason == "BUL" || reason == "RIV") && t.third != dpsReasonEnum)
+    return t.first != role1 || t.second != role2 // too strict: || ((reason == "BUL" || reason == "RIV") && t.third != dpsReasonEnum)
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt:286]: Leaving out the test on the dps reason as this is not important according to Mike W

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsResource.kt [hunk 4]
```diff
@@ -61,6 +61,7 @@ class NonAssociationsResource(
             results.associate { "${it.id}" to "nomis=${it.nomisNonAssociation}, dps=${it.dpsNonAssociation}" }
           telemetryClient.trackEvent("non-associations-reports-reconciliation-success", map)
           log.info("Non-associations reconciliation report logged")
+          log.info("Non-associations reconciliation temp map = $map")
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsResource.kt:64]: whoops - this should not be in, will remove in a later PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/NonAssociationsApiMockServer.kt [hunk 5]
```diff
@@ -39,33 +40,20 @@ class NonAssociationsApiMockServer : WireMockServer(WIREMOCK_PORT) {
 
   fun stubHealthPing(status: Int) {
     stubFor(
-      get("/health/ping").willReturn(
-        aResponse()
-          .withHeader("Content-Type", "application/json")
-          .withBody(if (status == 200) "pong" else "some error")
-          .withStatus(status),
-      ),
+      get("/health/ping").willReturn(okJson(if (status == 200) "pong" else "some error")),
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/wiremock/NonAssociationsApiMockServer.kt:43]: switched to use convenience funs

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt (hunk 3 above)
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt:286]: This comment makes sense for this commit but will it mean anything in the future? Maybe better to remove the comment from the code but add it to the Pull Request?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationServiceTest.kt [hunk 6]
```diff
@@ -83,7 +83,89 @@ class NonAssociationsReconciliationServiceTest {
...
+    )
+
+    assertThat(nonAssociationsReconciliationService.checkMatch(NonAssociationIdResponse(OFFENDER1, OFFENDER2)))
+      .asList()
+      .isEmpty()
+  }
+  /*
+  nomisNonAssociation=NonAssociationReportDetail(type=LAND, createdDate=2022-01-01, expiryDate=2022-01-01, closed=null, roleReason=VIC, roleReason2=PER, dpsReason=, comment=comment1),
+  dpsNonAssociation=NonAssociationReportDetail(type=LANDING, createdDate=2022-01-01T10:00:00, expiryDate=, closed=true, roleReason=NOT_RELEVANT, roleReason2=NOT_RELEVANT, dpsReason=BULLYING, comment=comment1))]
+   */
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationServiceTest.kt:103]: Not sure why this is here?

**Review by mikehalmamoj — APPROVED** [review]

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationServiceTest.kt (hunk 6 above)
Comment (steverendell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationServiceTest.kt:103]: will remove!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt (hunk 3 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsReconciliationService.kt:286]: I suspect I may need to remove the role checks too depending on what happens when this runs in preprod, so Ill revisit

**Pull request merged**